addons/l10n_th/views/report_invoice.xml

```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document" primary="True">
+        <xpath expr="//div[@name='address_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
```

Comment:
https://github.com/odoo/odoo/blob/30a8e933b3cda9c3c663c4d05d1efdc023b17b1e/addons/account/views/report_invoice.xml#L28-L29

The div you are inheriting from is displayed only if the shipping address of the invoice is the same as the partner's official address.
So if that is not the case, the 'Headquarter'/'Branch' of the customer will not be displayed in the report, which seems a bug to me.